gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -293,7 +301,18 @@ public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Storage storage
       }
     }
 
-    this.storageRequestAuthorizer = initializeStorageRequestAuthorizer(storageOptions);
+    this.storageRequestAuthorizer = initializeStorageRequestAuthorizer(this.storageOptions);
+    this.shouldAuthenticationPerRequest = this.storageOptions.getShouldAuthenticatePerRequest();
+
+    if (this.storageOptions.getAccessTokenProvider() != null) {
+      try {
+        this.accessTokenProvider = this.storageOptions.getAccessTokenProvider().getDeclaredConstructor().newInstance();
```

Comment:
In Delegation token support, the access token provider comes from the delegation token instead of directly through configuration, so the implementation here won't work for delegation tokens. Could you just inject the access token provider directly into GoogleCloudStorageImpl, if some configuration is set to "true"?

Reply:
This is fixed now so that `GoogleCloudStorageImpl` keeps track of the `AccessTokenProvider` to provide the AccessToken.